create_bot_knowledge.py

```diff
@@ -80,139 +82,243 @@ def connect_to_source(project: Tuple[str, str]):
+    }
+
+    # TODO: think about saving comments
+    # would it be valuable?
+
+
+def issues_analysis(project: Repository, project_knowledge: Path):
+    """Analysis of every closed issue in repository
+    
+    Arguments:
+        project {Repository} -- currently the PyGithub lib is used because of its functionality
+                                ogr unfortunatelly did not provide enough to properly analyze issues
+        
+        project_knowledge {Path} -- project directory where the issues knowledge will be stored
+    """
+    _LOGGER.info('-------------Issues (that are not PR) Analysis-------------')
+    data_path = project_knowledge.joinpath('./issues.json')
+
+    data = load_previous_knowledge(data_path)
+    current_issues = [issue for issue in project.get_issues(state='closed') if issue.pull_request is None]
+    new_issues = get_only_new_entities(data, current_issues)
+    
+    if len(new_issues) == 0:
+        return
+
+    for idx, issue in enumerate(new_issues, 1):
+        _LOGGER.info("Analysing ISSUE no. %d/%d" % (idx, len(new_issues)))
+        analyze_issue(issue, data)
+
+    save_knowledge(data_path, data)
+
+
+def analyze_pull_request(pull: PullRequest, results: Dict[str, Dict[str, Union[Optional[str], float]]]):
+    """Analyse pull request and save its desired features to results."""
+    commits = pull.commits
+    # TODO: Use commits to extract information.
+    # commits = [commit for commit in pull.get_commits()]
+
+    created_at = pull.created_at.timestamp()
+    closed_at = pull.closed_at.timestamp()
+
+    # Get the review approvation if it exists
+    # approvation = next((review for review in reviews if review.state == 'APPROVED'), None)
+    # pr_approved = approvation.submitted_at.timestamp() if approvation is not None else None
+    # pr_approved_by = pull.approved_by.name if approvation is not None else None
+    # time_to_approve = pr_approved - created_at if approvation is not None else None
+
+    merged_at = pull.merged_at.timestamp() if pull.merged_at is not None else None
+    
+    labels = [label.name for label in pull.get_labels()]
+
+    results[str(pull.number)] = {
+        "size": get_labeled_size(labels),
+        "labels": get_non_standalone_labels(labels),
+        "created_by": pull.user.login,
+        "created_at": created_at,
+        #"approved_at": pr_approved,
+        #"approved_by": pr_approved_by,
+        #"time_to_approve": time_to_approve,
+        "closed_at": closed_at,
+        "closed_by": pull.as_issue().closed_by.login,
+        "time_to_close": closed_at - created_at,
+        "merged_at": merged_at,
+        "commits_number": commits, 
+https://trello.com/c/VRyizGlU/846-data-collection-from-github-repositories-nokr
+https://trello.com/c/VRyizGlU/846-data-collection-from-github-repositories-nokrts()),
+https://trello.com/c/VRyizGlU/846-data-collection-from-github-repositories-nokr
+https://trello.com/c/VRyizGlU/846-data-collection-from-github-repositories-nokr
```

Comment:
Not sure what is this :)

Reply:
what the hell is this :D embarrassing, will remove this